api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,102 @@
+import { Script } from "vm";
+import { Config } from "@jest/types";
+import { execSync } from "child_process";
+import commander from "commander";
+import { LedgerRunner } from "../lib/ledgers/ledger_runner";
+import { CndRunner } from "../lib/cnd/cnd_runner";
+import { HarnessGlobal } from "../lib/utils";
+import NodeEnvironment from "jest-environment-node";
+
+// ************************ //
+// Setting global variables //
+// ************************ //
+
+declare var global: HarnessGlobal;
+
+interface ConfigInterface {
+    ledgers: string[];
+}
+
+export class E2ETestEnvironment extends NodeEnvironment {
+    private testPath: string;
+    private docblockPragmas: string;
+    private projectRoot: string;
+    private testRoot: string;
+    private logDir: string;
+    private ledgerRunner: LedgerRunner;
+    constructor(config: Config.ProjectConfig, context: any) {
+        super(config);
+        this.testPath = context.testPath;
+        this.docblockPragmas = context.docblockPragmas;
+
+        this.projectRoot = execSync("git rev-parse --show-toplevel", {
+            encoding: "utf8",
+        }).trim();
+        this.testRoot = this.projectRoot + "/api_tests";
+        this.logDir = this.projectRoot + "/api_tests/log";
+
+        this.ledgerRunner = new LedgerRunner(this.projectRoot, this.logDir);
+        // this.nodeRunner = new CndRunner(this.projectRoot, this.logDir);
+    }
+
+    async setup() {
+        await super.setup();
+        console.log(`Starting up test environment`);
+
+        // setup global variables
+        this.global.projectRoot = this.projectRoot;
+        this.global.testRoot = this.testRoot;
+        this.global.logRoot = this.logDir;
+        this.global.ledgerConfigs = {};
+        this.global.verbose = true; // TODO change default to false
+        if (commander.verbose) {
+            this.global.verbose = true;
+        }
+
+        // Will trigger if docblock contains @ledgers
+        // @ts-ignore
+        const configString = this.docblockPragmas.ledgers;
+        if (configString) {
+            const config: ConfigInterface = JSON.parse(configString);
```

Comment:
Not sure how well this fits into "10-50%", it does concern the "API" the tests are using so I am gonna mention it:

I find the use of JSON here a bit awkward, why not just a comma separated list? Seems burdensome to have the test author write JSON in the doc comment.

Reply:
Fair point. I had this because I wanted to configure `actors` as well which I removed later on again as we don't need it.

Reply:
I opted for adding it again in order to be able to configure the log folder name.

Reply:
Okay, it is good enough for a starting point.
From reading the docs, I suspect thought that the idea is that you use different pragmas for different parameters, kind of like:

```js
/*
 * @logFolder e2e
 * @ledgers bitcoin, ethereum
 */

```

Reply:
@bonomat I unresolved this because I added a (non-blocking) comment.

Reply:
Works :)